lib/elixir/lib/calendar/iso.ex

```diff
@@ -953,10 +1007,12 @@ defmodule Calendar.ISO do
   """
   @doc since: "1.5.0"
   @impl true
-  @spec valid_date?(year, month, day) :: boolean
-  def valid_date?(year, month, day) do
-    month in 1..12 and year in -9999..9999 and
-      (is_integer(day) and day >= 1 and day <= days_in_month(year, month))
+  @spec valid_date?(year :: integer, month :: integer, day :: integer) :: boolean
+  def valid_date?(year, month, day)
+      when is_integer(year) and is_integer(month) and is_integer(day) do
+    year >= -9999 and year <= 9999 and
+      month >= 1 and month <= 12 and
+      day >= 1 and day <= days_in_month(year, month)
```

Comment:
I think it's fine as is, but this would be slightly shorter:

```elixir
year in -9999..9999 and
  month in 1..12 and
  day in 1..days_in_month(year, month)
```

Reply:
I love the way you used it in a range. that's great. thank you @wojtekmach